src/library_egl.js

```diff
@@ -289,7 +370,16 @@ var LibraryEGL = {
       return 0; /* EGL_NO_CONTEXT */
     }
 
-    _glutInitDisplayMode(0xB2 /* GLUT_RGBA | GLUT_DOUBLE | GLUT_DEPTH | GLUT_MULTISAMPLE | GLUT_STENCIL */);
+    EGL.currentEGLConfig = config;
+
+    var displayMode = (0x0002 /* GLUT_RGBA|GLUT_DOUBLE */ | 
+      (EGL.configAttribs[config-1].alpha && 0x0008) /* GLUT_ALPHA */ |
+      (EGL.configAttribs[config-1].depth && 0x0010) /* GLUT_DEPTH */ |
+      (EGL.configAttribs[config-1].stencil && 0x0020) /* GLUT_STENCIL */ |
+      (EGL.configAttribs[config-1].buffers && 0x0080) /* GLUT_MULTISAMPLE */
+    )
+    
+    _glutInitDisplayMode(displayMode);
```

Comment:
The reason why this code path calls to GLUT to initialize the GL context is historical since back then we didn't have a good other API to create a context, and in the long run we certainly don't want to do this anymore.

Since then we've added a "native" WebGL context creation function `GL.createContext`, which is available here: https://github.com/kripken/emscripten/blob/master/src/library_gl.js#L405. See here for an example of how that is called: https://github.com/kripken/emscripten/blob/master/src/library_html5.js#L1726. If you're interested to doing this change, feel free to, although certainly can leave that separate for later as well.